docs/src/main/asciidoc/kms.adoc

```diff
@@ -0,0 +1,74 @@
+
+Spring Cloud GCP offers a utility template class `KmsTemplate` which allows you to conveniently specify secrets for encryption/decryption.
+
+=== Dependency Setup
+
+To begin using this library, add the `spring-cloud-gcp-starter-vision` artifact to your project.
+
+Maven coordinates, using <<getting-started.adoc#_bill_of_materials, Spring Cloud GCP BOM>>:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>spring-cloud-gcp-starter-kms</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+  implementation("com.google.cloud:spring-cloud-gcp-starter-kms")
+}
+----
+
+=== Configuration
+
+The following options may be configured with Spring Cloud GCP KMS libraries.
+
+|===========================================================================
+| Name | Description | Required | Default value
+| `spring.cloud.gcp.kms.enabled` | Enables or disables Google Cloud KMS autoconfiguration | No | `true`
+| `spring.cloud.gcp.kms.project-id` | GCP project ID of the project using Cloud KMS APIs, if different from the one in the <<spring-cloud-gcp-core,Spring Cloud GCP Core Module>>. | No | Project ID is typically inferred from https://cloud.google.com/sdk/gcloud/reference/config/set[`gcloud`] configuration.
+| `spring.cloud.gcp.kms.credentials.location` | Credentials file location for authenticating with the Cloud KMS APIs, if different from the ones in the <<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> | No | Inferred from https://cloud.google.com/docs/authentication/production[Application Default Credentials], typically set by https://cloud.google.com/sdk/gcloud/reference/auth/application-default[`gcloud`].
+|===========================================================================
+
+
+=== Basic Usage
+
+Once you have added the `spring-cloud-gcp-starter-kms` to your project, the autoconfiguration class `com.google.cloud.spring.autoconfigure.kms.GcpKmsAutoConfiguration` will be activated for your project.
+
+The `com.google.cloud.spring.kms.KmsTemplate` class provided by the autoconfiguration is the entrypoint to using Spring Cloud GCP support for Google KMS. This class allows you to specify a Cloud KMS key in your project via a URI string (format described below) and perform encryption/decryption with it.
+
+The template class automatically validates the CRC32 checksums received responses from Cloud KMS APIs to verify correctness of the response.
+
+==== Cloud KMS URI format
+
+Spring Cloud GCP offers the following URI syntax to specify Cloud KMS keys in your project:
+
+[source]
+----
+ 1. Shortest form - specify the key by location ID, key ring ID, and key ID.
+ The project is inferred from the spring.cloud.gcp.kms.project-id if set, otherwise
+ the default GCP project (such as using application-default-credentials) is used.
+
+ kms://{location-id}/{key-ring-id}/{key-id}
```

Comment:
What do you think of an even shorter form for global keys?
kms://{key-ring-id}/{key-id}

Reply:
Great Idea, will do it!

Reply:
Cool! I also think it could be more user-friendly to change the sample to instruct the users to create a specific key-id, and use this short version of it to refer to it.

Similarly, we could modify the integration tests to do the same, so that they are portable across GCP projects. However, we would have to get rid of the tests that rely on a specific ciphertext literal.

Reply:
Changed it! Will also change also the sample.  

```
However, we would have to get rid of the tests that rely on a specific ciphertext literal.
```

For the ones mocked it is covered due to mocking the response. On the Integration tests I avoided hardcoding the ciphertexts and instead used encrypt/decrypt together  so anyone could run the tests.

Reply:
Updated also the sample.